docs/standard-library/vectorized-stl-algoritms.md

```diff
@@ -0,0 +1,87 @@
+---
+description: "Vectorized STL Algorithms"
+title: "Vectorized STL Algorithms"
+ms.date: "09/19/2025"
+helpviewer_keywords: ["Vector Algorithms", "Vectorization", "SIMD"]
+---
+# Vectorized STL Algorithms
+
+Under certain conditions, STL algorithms execute not element-wise, but multiple element at once on a single CPU core. This is possible due to SIMD (single instruction, multiple data). The use of such approach instead of 
+element-wise approach is called vectorization. The implementation that is not vectorized is called scalar.
+
+The conditions for vectorization are:
+ - The container or range is contigous. `array`, `vector`, and `basic_string` are contigous containers, `span` and `basic_string_view` provide conditions ranges.
+ - There are such SIMD insstructions available for the target platform that implement the particular algorithm on particular element types efficiently. Usually this is true for plain types (like built-in integers) and simple operations.
+ - Either of the following:
+     - The compiler is capable emiting vectorized machine code for an implementation written as scalar code (auto-vectorization)
+     - The implementation itself is written as vectorized code (manual vectorization)
+
+## Auto-vectorization in STL
+
+See [Auto-Vectorizer](../parallel/auto-parallelization-and-auto-vectorization.md#auto-vectorizer). It applies to the STL implementation code the same way as to user code.
+
+Algorithms like `transform`, `reduce`, `accumulate` heavily benefit from auto-vectorization.
+
+## Manual vectorization in STL
+
+For x64 and x86 targets, certain algorithms have manual vectorization implemented. This implementation is pre-compiled, and uses runtime CPU dispatch, so it is engaged on suitable CPUs only.
+
+The manually vectorized algorithms use template meta-programming to detect the suitable element types, so they only vectorized for simple types, like standard integer types.
+
+Generally, programs either benefit in performance from this manual vectorization or are unaffected by it. In case of any problem, you can disable manual vectorization by defining `_USE_STD_VECTOR_ALGORITHMS` macro set to 0.
+
+The following algorithms have manual vectorization controlled via `_USE_STD_VECTOR_ALGORITHMS` macro:
```

Comment:
Having such a list kinda makes me nervous; if someone vectorizes a new function and forgets adding it here, we've got problems. This is very easy when it isn't even in the ms-stl repo.

Also what kind of sorting is that

Reply:
I know,
Plus it is accurate for like the latest Preview, not latest production.
But I still think providing an approximation here is useful and better than no list.